www/docs/server/infer-types.md

```diff
@@ -171,3 +171,99 @@ function usePostById(input: PostByIdInput, options?: PostByIdOptions) {
   return trpc.post.byId.useQuery(input, options);
```

Comment:
Since this is part of the react-query library, should we actually move these additions to a new file under the react docs? Perhaps `www/docs/react/infer-types.md`?